app/Party.php

```diff
@@ -445,94 +580,11 @@ public function scopeUpcomingEventsInUserArea($query, $user)
-            ->whereDate('event_date', '<', date('Y-m-d'))
-            ->orderBy('event_date', 'DESC');
-    }
-
-    /**
-     * [scopeUsersPastEvents description]
-     * Get all Past Events from the User or User's groups, or any Past Events
-     * in which the User(s) has attended but is NOT a part of the
-     * group associated.
-     * @author Christopher Kelker
-     * @date   2019-05-28T09:50:22+010
-     * @param  [type]                  $query
-     * @param  [type]                  $user_ids
-     * @return [type]
-     */
-    public function scopeUsersPastEvents($query, array $user_ids = null)
-    {
-        // if no $user_ids are supplied, then use the current Auth's ID
-        if (empty($user_ids)) {
-            $user_ids[] = auth()->id();
-        }
-
-        return $query->join('groups', 'groups.idgroups', '=', 'events.group')
-            ->join('users_groups', 'users_groups.group', '=', 'groups.idgroups')
-            ->join('events_users', 'events_users.event', '=', 'events.idevents')
-            ->whereNotNull('events.wordpress_post_id')
-            ->whereDate('events.event_date', '<', date('Y-m-d'))
-            // Not left the group.
-            ->whereNull('users_groups.deleted_at')
-
-            ->where(function ($query) use ($user_ids) {
-                $query->whereIn('users_groups.user', $user_ids)
-                    ->orWhereIn('events_users.user', $user_ids);
-            })
-
-            ->select('events.*')
-            ->groupBy('idevents')
-            ->orderBy('events.event_date', 'DESC');
-    }
-
-    /**
-     * [scopeUsersUpcomingEvents description]
-     *
-     * Get all Upcoming Events from the User or User's groups
-     *
-     * @param  [type]                  $query
-     * @param  [type]                  $user_ids
-     * @return [type]
-     */
-    public function scopeUsersUpcomingEvents($query, array $user_ids = null)
-    {
-        // if no $user_ids are supplied, then use the current Auth's ID
-        if (empty($user_ids)) {
-            $user_ids[] = auth()->id();
-        }
-
-        return $query->join('groups', 'groups.idgroups', '=', 'events.group')
-            ->leftJoin('users_groups', 'users_groups.group', '=', 'groups.idgroups')
-            ->leftJoin('events_users', 'events_users.event', '=', 'events.idevents')
-            ->whereNotNull('events.wordpress_post_id')
-            ->where('users_groups.status', 1)
-            ->whereNull('users_groups.deleted_at')
-            ->whereDate('events.event_date', '>=', date('Y-m-d'))
-
-            ->where(function ($query) use ($user_ids) {
-                $query->whereIn('users_groups.user', $user_ids)
-                    ->orWhereIn('events_users.user', $user_ids);
-            })
-
-            ->select('events.*')
-            ->groupBy('idevents')
-            ->orderBy('events.event_date', 'DESC');
+        $query = $query->future();
+        $query = $query->whereNull('wordpress_post_id');
```

Comment:
Wonder could this use the new 'approved' scope?  Can scopes be magically negated?  e.g. $query->future()->not()->approved() or somesuch?

Reply:
It looks from https://stackoverflow.com/questions/34645129/laravel-negate-local-scope as though there isn't magic to do this.